### PR TITLE
localized diary topics

### PR DESCRIPTION
**Pull request by ColdEvul**
**When merged this pull request will:**
- Removed date topic from the diary records (New function from [indev](https://community.bistudio.com/wiki/createDiaryRecord))
- Added localized topic using pre existing localization strings.

### CODE REVIEW

File: addons/help/XEH_postInit.sqf [hunk 1]
```diff
@@ -37,7 +37,7 @@ if (!hasInterface) exitWith {};
     // delete last line breaks
     _keys = _keys select [0, count _keys - 10];
 
-    _unit createDiaryRecord [QGVAR(docs), [localize "STR_CBA_Help_Keys", _keys]];
-    _unit createDiaryRecord [QGVAR(docs), [localize "STR_CBA_Credits", call (uiNamespace getVariable QGVAR(credits))]];
-    _unit createDiaryRecord [QGVAR(docs), [localize "STR_CBA_Addons", call (uiNamespace getVariable QGVAR(mods))]];
+    _unit createDiaryRecord [QGVAR(docs), [localize "STR_CBA_Help_Keys", format["<font size=20>%1</font><br/>%2", localize "STR_CBA_Help_Keys", _keys], taskNull, "", false]];
+    _unit createDiaryRecord [QGVAR(docs), [localize "STR_CBA_Credits", format["<font size=20>%1</font><br/>%2", localize "STR_CBA_Credits", call (uiNamespace getVariable QGVAR(credits))]], taskNull, "", false];
+    _unit createDiaryRecord [QGVAR(docs), [localize "STR_CBA_Addons", format["<font size=20>%1</font><br/>%2", localize "STR_CBA_Addons", call (uiNamespace getVariable QGVAR(mods))]], taskNull, "", false];
```
Comment (commy2) [inline addons/help/XEH_postInit.sqf]:
```suggestion
    _unit createDiaryRecord [QGVAR(docs), [localize "STR_CBA_Addons", format ["<font size=20>%1</font><br/>%2", localize "STR_CBA_Addons", call (uiNamespace getVariable QGVAR(mods))]], taskNull, "", false];
```

File: addons/help/XEH_postInit.sqf [hunk 2]
```diff
@@ -37,7 +37,7 @@ if (!hasInterface) exitWith {};
     // delete last line breaks
     _keys = _keys select [0, count _keys - 10];
 
-    _unit createDiaryRecord [QGVAR(docs), [localize "STR_CBA_Help_Keys", _keys]];
-    _unit createDiaryRecord [QGVAR(docs), [localize "STR_CBA_Credits", call (uiNamespace getVariable QGVAR(credits))]];
-    _unit createDiaryRecord [QGVAR(docs), [localize "STR_CBA_Addons", call (uiNamespace getVariable QGVAR(mods))]];
+    _unit createDiaryRecord [QGVAR(docs), [localize "STR_CBA_Help_Keys", format["<font size=20>%1</font><br/>%2", localize "STR_CBA_Help_Keys", _keys], taskNull, "", false]];
+    _unit createDiaryRecord [QGVAR(docs), [localize "STR_CBA_Credits", format["<font size=20>%1</font><br/>%2", localize "STR_CBA_Credits", call (uiNamespace getVariable QGVAR(credits))]], taskNull, "", false];
```
Comment (commy2) [inline addons/help/XEH_postInit.sqf]:
```suggestion
    _unit createDiaryRecord [QGVAR(docs), [localize "STR_CBA_Credits", format ["<font size=20>%1</font><br/>%2", localize "STR_CBA_Credits", call (uiNamespace getVariable QGVAR(credits))]], taskNull, "", false];
```

File: addons/help/XEH_postInit.sqf [hunk 3]
```diff
@@ -37,7 +37,7 @@ if (!hasInterface) exitWith {};
     // delete last line breaks
     _keys = _keys select [0, count _keys - 10];
 
-    _unit createDiaryRecord [QGVAR(docs), [localize "STR_CBA_Help_Keys", _keys]];
-    _unit createDiaryRecord [QGVAR(docs), [localize "STR_CBA_Credits", call (uiNamespace getVariable QGVAR(credits))]];
-    _unit createDiaryRecord [QGVAR(docs), [localize "STR_CBA_Addons", call (uiNamespace getVariable QGVAR(mods))]];
+    _unit createDiaryRecord [QGVAR(docs), [localize "STR_CBA_Help_Keys", format["<font size=20>%1</font><br/>%2", localize "STR_CBA_Help_Keys", _keys], taskNull, "", false]];
```
Comment (commy2) [inline addons/help/XEH_postInit.sqf]:
```suggestion
    _unit createDiaryRecord [QGVAR(docs), [localize "STR_CBA_Help_Keys", format ["<font size=20>%1</font><br/>%2", localize "STR_CBA_Help_Keys", _keys], taskNull, "", false]];
```

**Comment by commy2** [comment]
How to turn a draft into a pull request?

Does this error with current release?

**Comment by dedmen** [comment]
>How to turn a draft into a pull request?

firefox_2019-06-12_09-42-42

**Comment by dedmen** [comment]
While at this, people have been asking for a way to hide diary records
https://forums.bohemia.net/forums/topic/223694-remove-diary-records-from-mods-or-others-already-there/?tab=comments#comment-3361406
Maybe add a setting or variable to disable creation of it?

**Comment by ColdEvul** [comment]
Will make the adjustments digested above
and while at it; I can make a setting to disable the records as well for this pull request.

> Does this error with current release?

Higly likely; don't know to be honest. Was also the reason why I made a draft of it.

**Comment by commy2** [comment]
>digested

Suggested? Please don't eat the pull request.

**Comment by ColdEvul** [comment]
haha phone autocorrect

File: addons/help/XEH_postInit.sqf (hunk 3 above)
Comment (ColdEvul) [reply addons/help/XEH_postInit.sqf]: one of the `]` was also wrong at the end. I missed it and the error in testing.

**Comment by ColdEvul** [comment]
image

**Comment by ColdEvul** [comment]
Setting is local.

**Comment by ColdEvul** [comment]
All done ready for review.

I am unsure if the setting handeling is a good solution. But the postInit did not handle anything else so no point running the rest.

**Comment by commy2** [comment]
Not a fan of making a setting for this. It seems to me that the overhead of a setting, even though small, is not worth such a minor thing as some briefing entries.
The guy on the BIF even said that he had (and lost) a solution in the past that did just mission stuff to hide the diary entries. I would prefer if we just rediscover that solution and put it on the wiki.

**Comment by ColdEvul** [comment]
Well I also removed all diaries once as well in a super hacky way but also lost that solution.

Tho there will be 2 new commands that allow scripts to remove diary records (removeDiarySubject, rem﻿oveDiaryRecor﻿﻿d). I stated this on the forum post as well.

**Comment by commy2** [comment]
That settles it then. I'll doc it on the wiki. Setting not needed.

**Comment by ColdEvul** [comment]
Settings reverted

**Comment by commy2** [comment]
Remove will be:
```sqf
player removeDiarySubject "cba_help_docs";
```

or:
```sqf
player removeDiaryRecord ["cba_help_docs", localize "STR_CBA_Help_Keys"];
player removeDiaryRecord ["cba_help_docs", localize "STR_CBA_Credits"];
player removeDiaryRecord ["cba_help_docs", localize "STR_CBA_Addons"];
```
in next Arma stable I think.

**Comment by ColdEvul** [comment]
Seams not to be working properly after the update

**Comment by commy2** [comment]
What? This pull request?

**Comment by ColdEvul** [comment]
Unknown tbh. The extra parameters don't seam to work on one of my branches were I'm running them.

**Comment by commy2** [comment]
What do you mean unknown? If you mean the commands I posted,
https://community.bistudio.com/wiki/removeDiaryRecord and
https://community.bistudio.com/wiki/removeDiarySubject
are not in 1.94.

**Comment by ColdEvul** [comment]
oh no i'm talking about the new params introduced to `createDiaryRecord`

**Comment by commy2** [comment]
I suppose the changes to
https://community.bistudio.com/wiki/createDiaryRecord
are not in 1.94 either. The wiki still says "dev".

This command:
https://community.bistudio.com/wiki/addWeaponWithAttachmentsCargoGlobal
for example says 1.94. But it may just be the wiki being outdated. 

I think these changes didn't make it in 1.94

**Comment by ColdEvul** [comment]
Yeah seams like it did not. :(

**Comment by commy2** [comment]
```sqf
0 spawn {
    waitUntil {!isNil "cba_help_DiaryRecordAddons"};

    // delete whole subject
    player removeDiarySubject "cba_help_docs";

    // or delete individual records
    player removeDiaryRecord ["cba_help_docs", cba_help_DiaryRecordKeys];
    player removeDiaryRecord ["cba_help_docs", cba_help_DiaryRecordCredits];
    player removeDiaryRecord ["cba_help_docs", cba_help_DiaryRecordAddons];
};
```

**Comment by commy2** [comment]
fuck this shit

**Pull request closed without merging**